File: core/LianBiaoZhongDaoShuDiKgeJieDianLcof/LianBiaoZhongDaoShuDiKgeJieDianLcof.java
package com.dbc;

public class LianBiaoZhongDaoShuDiKgeJieDianLcof {
    public class ListNode {
        int val;
        ListNode next;

        ListNode(int x) {
            val = x;
        }
    }

    public ListNode getKthFromEnd(ListNode head, int k) {
        ListNode preHead = new ListNode(0);
        preHead.next = head;
        while (k > 0) {
            head = head.next;
            k--;
        }
        while (head != null) {
            preHead = preHead.next;
            head = head.next;
        }
        return preHead.next;
    }
}

File: core/LianBiaoZhongDaoShuDiKgeJieDianLcof/LianBiaoZhongDaoShuDiKgeJieDianLcof.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


def getKthFromEnd(head: ListNode, k: int) -> ListNode:
    pre_head = ListNode()
    pre_head.next = head
    while k > 0 and head:
        head = head.next
        k -= 1

    while head:
        head = head.next
        pre_head = pre_head.next

    return pre_head.next




